doc/specification.md

```diff
@@ -0,0 +1,52 @@
+# Test Specification
+
+This document describes how to use the JSON test specifications to write your own test executor for
+[`wasi-testsuite`](..). The test executor included in this project provides a completely-usable
+reference implementation, but projects with other requirements may want to run the tests in their
+own way (e.g., no Python dependency). The JSON test specifications provide a simple way to verify
+that the tests indeed passed.
+
+### Configuration
+
+Before executing anything, a test executor is expected to:
+- find all `*.wasm` and `*.wat` files in a given subdirectory &mdash; these are the _test cases_
```

Comment:
I think it's just `*.wasm` at the moment. See https://github.com/WebAssembly/wasi-testsuite/blob/main/test-runner/wasi_test_runner/test_suite_runner.py#L38

Is there any usecase for `*.wat` files? I don't know about Wasmtime specifically, but I know some of the runtimes can't accept `.wat` files, so the test runtime would have to compile them beforehand. I'd personally avoid that unless we have a strong usecase for it.

Reply:
I just recall that in the wasi-threads tests some of them were `*.wat` so I thought that was allowed. I'll remove it for now but it does make things easier to have the `*.wat` files around when troubleshooting a failing test &mdash; saves a step.